File: android/SmartIRHub/app/src/main/java/com/ms8/smartirhub/android/_tests/dev_playground/remote_layout/asymmetricgridview/AsymmetricViewHolder.java
package com.ms8.smartirhub.android._tests.dev_playground.remote_layout.asymmetricgridview;

import android.view.View;
import androidx.recyclerview.widget.RecyclerView;

public class AsymmetricViewHolder<VH extends RecyclerView.ViewHolder>
    extends RecyclerView.ViewHolder {
  final VH wrappedViewHolder;

  public AsymmetricViewHolder(VH wrappedViewHolder) {
    super(wrappedViewHolder.itemView);
    this.wrappedViewHolder = wrappedViewHolder;
  }

  public AsymmetricViewHolder(View view) {
    super(view);
    wrappedViewHolder = null;
  }
}

File: android/SmartIRHub/app/src/main/java/com/ms8/smartirhub/android/_tests/dev_playground/remote_layout/asymmetricgridview_k/AsymmetricViewHolder.kt
package com.ms8.smartirhub.android._tests.dev_playground.remote_layout.asymmetricgridview_k

import android.view.View
import androidx.recyclerview.widget.RecyclerView

class AsymmetricViewHolder<VH : RecyclerView.ViewHolder> : RecyclerView.ViewHolder {
    internal val wrappedViewHolder: VH?

    constructor(wrappedViewHolder: VH) : super(wrappedViewHolder.itemView) {
        this.wrappedViewHolder = wrappedViewHolder
    }

    constructor(view: View) : super(view) {
        wrappedViewHolder = null
    }
}